drivers/input/touchscreen/ef1e_tp/ef1e_tp_driver.c

```diff
@@ -641,6 +641,29 @@ static int ef1e_tp_remove(struct platform_device *dev)
 	return 0;
 }
 
+static void ef1e_dp_shutdown(struct platform_device *pdev)
```

Comment:
The resources would be double-freed here as they have been released in the `.remove` callback.

What's the purpose of preferrence of `.shutdown` over `.remove`?

Reply:
Remove free resources. S5 stability test call shutdown, don`t call remove.